### ISSUE TITLE
Stats Date Picker : Localize example date format

### ISSUE THREAD

**Issue by annacmc** [Automattic/wp-calypso#83148]
Currently the date range selector heading shows an example date format, eg `Date Range (dd/mm/yyyy)`

This format will change with localization, so needs to be adjusted to display dynamically for other locales using different date formats. 

![Image](https://github.com/Automattic/wp-calypso/assets/30754158/4c2b99c5-a518-4403-8c0a-ea832e140bda)

**Comment by a8ck3n** [comment]
It looks like we can query Moment.js for the string formats in the current locale. For example, if we do the following:

```
const currentLocaleData = moment.localeData();
console.log( currentLocaleData.longDateFormat( 'L' ) );
```

This logs `MM/DD/YYYY`. Unfortunately, that doesn't match my system settings of `YYYY-MM-DD`. In my research, I'm not seeing any way to look up the users preferred date formats. Not via Moment.js nor via the JS Date and Intl classes.

If I inspect the `currentLocaleData` it says my locale is "en". It doesn't say "en-CA" which is my actual locale. So I'm thinking maybe it's working from a simplified list of locales for this reporting?

One interesting thing I discovered though is an empty date field will show the users preferred date format. Or at least in my case, it picks up the setting from my Mac and changing the setting is reflected by the date input. So that gives us two possible options here:

1. We remove the example format string from the control. This was useful initially because we were not using the "date" type on our input fields. Now that we are using the "date" type, the input is more constrained and obvious so maybe the example isn't needed.
2. Maybe a bit of a hack but we could use a disabled date input field with an empty value, then style it to look like info text. That would pick up the correct formatting from the system.

Thoughts on those @annacmc @grzegorz-cp ?

**Comment by a8ck3n** [comment]
I did a little test of option #2 and it kinda works in Chrome. I'm not confident it would behave correctly across browsers though. Plus I can't seem to take it out of the tab loop.

<img width="512" alt="SCR-20231019-poje" src="https://github.com/Automattic/wp-calypso/assets/40267301/999ccf82-74cd-426d-a313-880b463e6e80">

**Comment by a8ck3n** [comment]
cc: @madebynoam — Any thoughts on removing that string from the heading?

**Comment by kangzj** [comment]
Hi @a8ck3n Nice sleuthing!

I believe the locale moment is using is user's locale on wordpress.com, not the one on the local machine - we initialized moment with the setting.

I have a fix up: https://github.com/Automattic/wp-calypso/pull/83272/ <- tricky I know, but it works 😄

**Comment by a8ck3n** [comment]
> I have a fix up: #83272 <- tricky I know, but it works 😄

Just wanted to note that this doesn't work for me. The logic seems straight-forward enough but this is what I see now that the changes have been merged.

<img width="595" alt="SCR-20231025-msnu" src="https://github.com/Automattic/wp-calypso/assets/40267301/57f2ddcf-831d-48a9-a648-aea7b2020936">

I'll poke around and see if I can find out why it fails for my setup.

cc: @kangzj
